docs/architecture/specs/Spec-DID.md

```diff
@@ -116,9 +116,93 @@ A DDO document is composed of standard DDO attributes:
 
 Asset metadata can be included as one of the objects inside the `"service"` array, with type `"metadata"`.
 
+
+#### Nevermined internal attributes
+
+Nevermined registers some internal attributes in the asset metadata. They allow to store information that makes easier to understand the meaning of an asset, their life-cycle and where they can be used. The internal attributes are recorded under the `_nvm` section located at the root of the DDO. The `_nvm` section is internal to Nevermined and cannot be changed by the user.
+
+The attributes included at that section are:
+
+* `userId` - The unique identifier of the user owning the DDO. This attribute will be used to authorize metadata changes.
+* `appId` - The unique identifier of the application where this DDO has a meaning.
+* `networks` - Array of the network ids (ethereum blockchains) where the asset is registered.
+* `versions` - Array representing the list of changes made to the DDO
+
+##### User Id
+
+It's stored under the attribute `userId`. It's the unique identifier representing an user. The user authorization of Metadata documents is built around that attribute and only the user authenticated with that `userId` will be able to modify the metadata
+
+The `userId` will be a `string` storing an identifier in `UUID` format.
+
+```json
+{
+	"userId": "dff40170-37fc-11ed-be5b-9984d9f9ec35"
+}
+```
+
+##### Application Id
+
+It's stored under the attribute `appId`. The application Id identifies an unique application. It helps to refer to the specific domain/application where the DDO belongs to. Specifying that attribute will allow to have the same metadata storage providing service for different applications, and each application will be able to filter by that `appId`.
+
+The `appId` will be a `string` storing an identifier in `UUID` format. 
+
+```json
+{
+	"appId": "acde070d-8c4c-4f0d-9d8a-162843c10333"
+}
+```
+
+##### Versions
+
+It's stored under the attribute `versions`. The `versions` list stores the reference to all the changes done to the Metadata document.
+
+The `versions` attribute will be an array of objects storing the sorted list of all the different changes.
+
+```json
+{
+	"versions": {
+		"1": {
+			"updated": "2020-01-01T19:13:24Z",
+			"checksum": "89328493849328493284932"
+		},
+		"2": {
+			"updated": "2021-02-21T20:13:24Z",
+			"checksum": "045328094852309483203443"
+		}
+	}
+}
+```
+
+##### Nevermined Config section Example
+
+Here we put together all the changes included at the `_nvm` section:
+
+```json
+{
+	"@context": "https://w3id.org/did/v1",
+	"id": "did:nv:0c184915b07b44c888d468be85a9b28253e80070e5294b1aaed81c2f0264e429",
+	"_nvm": {
+		"userId": "dff40170-37fc-11ed-be5b-9984d9f9ec35",
+		"appId": "acde070d-8c4c-4f0d-9d8a-162843c10333",
+		"versions": {
+			"1": {
+				"updated": "2020-01-01T19:13:24Z",
+				"checksum": "89328493849328493284932"
+			},
+			"2": {
+				"updated": "2021-02-21T20:13:24Z",
+				"checksum": "045328094852309483203443"
+			}
+		}
```

Comment:
I'm having some problems figuring out how to map this on elasticsearch. Should this instead be a list?
```javascript
"versions": [
  {
    "updated": "2020-01-01T19:13:24Z",
    "checksum": "89328493849328493284932"
  },
  {
    "updated": "2021-02-21T20:13:24Z",
    "checksum": "045328094852309483203443"
  }
]
```

or
```javascript
"versions": [
  {
    "id": 1,
    "updated": "2020-01-01T19:13:24Z",
    "checksum": "89328493849328493284932"
  },
  {
    "id": 2,
    "updated": "2021-02-21T20:13:24Z",
    "checksum": "045328094852309483203443"
  }
]
```

Reply:
Both options look good to me. I prefer the second one with the version id because we could use that to identify one unique version.